File: service/app/src/main/java/io/miragon/blueprint/adapter/inbound/operaton/CancelPolicyWorker.java
package io.miragon.blueprint.adapter.inbound.operaton;

import dev.bpmcrafters.processengine.worker.ProcessEngineWorker;
import dev.bpmcrafters.processengine.worker.Variable;
import io.miragon.blueprint.adapter.process.BikeLeasingProcessProcessApi.ServiceTasks;
import io.miragon.blueprint.application.port.inbound.CancelInsurancePolicyUseCase;
import io.miragon.blueprint.domain.leasing.ApplicationId;
import org.springframework.stereotype.Component;

/** Compensation handler for the insurance policy — consumes the {@code cancelPolicy} external task. */
@Component
public class CancelPolicyWorker {

    private final CancelInsurancePolicyUseCase useCase;

    public CancelPolicyWorker(CancelInsurancePolicyUseCase useCase) {
        this.useCase = useCase;
    }

    @ProcessEngineWorker(topic = ServiceTasks.CANCEL_POLICY)
    public void cancelPolicy(@Variable String applicationId) {
        useCase.cancelPolicy(ApplicationId.of(applicationId));
    }
}

File: service/app/src/main/kotlin/io/miragon/blueprint/adapter/inbound/operaton/CancelPolicyWorker.kt
package io.miragon.blueprint.adapter.inbound.operaton

import dev.bpmcrafters.processengine.worker.ProcessEngineWorker
import dev.bpmcrafters.processengine.worker.Variable
import io.miragon.blueprint.adapter.process.BikeLeasingProcessProcessApi.ServiceTasks
import io.miragon.blueprint.application.port.inbound.CancelInsurancePolicyUseCase
import io.miragon.blueprint.domain.leasing.ApplicationId
import org.springframework.stereotype.Component

/** Compensation handler for the insurance policy — consumes the `cancelPolicy` external task. */
@Component
class CancelPolicyWorker(
    private val useCase: CancelInsurancePolicyUseCase,
) {

    @ProcessEngineWorker(topic = ServiceTasks.CANCEL_POLICY)
    fun cancelPolicy(@Variable applicationId: String) {
        useCase.cancelPolicy(ApplicationId.of(applicationId))
    }
}
